### ISSUE TITLE
Implement Merge Sort

### ISSUE THREAD

**Issue by ayan-biswas0412** [ayan-biswas0412/AlgoDocs#47]
Implement Merge sort algorithm in various languages given below

- [x] C++
- [ ] C
- [x] Java
- [x] JavaScript
- [x] Python
- [ ] Golang
- [ ] Ruby
- [ ] C#
- [ ] R
- [ ] Swift
- [ ] PhP

Edit the file https://github.com/ayan-biswas0412/AlgoDocs/blob/main/docs/sorting/merge_sort.md

**Comment by amypritc** [comment]
Hi, I really think you should be including the definitions of both merge and merge sort in this example. I created a pull request with the python implementation of both merge and merge sort

**Comment by ayan-biswas0412** [comment]
> Hi, I really think you should be including the definitions of both merge and merge sort in this example - merge sort is quite difficult to understand if you only have half of the procedure. I created a pull request with the python implementation of both merge and merge sort.

yah sure

**Comment by amypritc** [comment]
Would it be okay to add OCaml? I can add a functional/recursive implementation of this algorithm using OCaml.

**Comment by ayan-biswas0412** [comment]
> Would it be okay to add OCaml? I can add a functional/recursive implementation of this algorithm using OCaml.

sure, go ahead

**Comment by ayan-biswas0412** [comment]
> Please assign me this issue. I will code the solution in C.

Yeah sure go ahead, your contribution will help this project to grow more.
